salt/log/setup.py

```diff
@@ -1088,6 +1259,69 @@ def patch_python_logging_handlers():
         logging.handlers.QueueHandler = QueueHandler
 
 
+def __process_multiprocessing_logging_zmq(opts, port):
+    # Avoid circular import
+    import salt.utils.process
+    salt.utils.process.appendproctitle('MultiprocessingLoggingQueue')
+
+    # Assign UID/GID of user to proc if set
+    from salt.utils.verify import check_user
+    user = opts.get('user')
+    if user:
+        check_user(user)
+
+    from salt.utils.platform import is_windows
+    if is_windows():
+        # On Windows, creating a new process doesn't fork (copy the parent
+        # process image). Due to this, we need to setup all of our logging
+        # inside this process.
+        setup_temp_logger()
+        setup_console_logger(
+            log_level=opts.get('log_level'),
+            log_format=opts.get('log_fmt_console'),
+            date_format=opts.get('log_datefmt_console')
+        )
+        setup_logfile_logger(
+            opts.get('log_file'),
+            log_level=opts.get('log_level_logfile'),
+            log_format=opts.get('log_fmt_logfile'),
+            date_format=opts.get('log_datefmt_logfile'),
+            max_bytes=opts.get('log_rotate_max_bytes', 0),
+            backup_count=opts.get('log_rotate_backup_count', 0)
+        )
+        setup_extended_logging(opts)
+
+    import traceback
+    context = zmq.Context()
+    puller = context.socket(zmq.PULL)
+    try:
+        puller.bind('tcp://127.0.0.1:{}'.format(port))
+    except Exception:
+        print("MP LOGGING TB %s" %('\n'.join(traceback.format_stack())))
+        raise
+    try:
+        while True:
+            try:
+                record = pickle.loads(puller.recv())
+                if record is None:
+                    # A sentinel to stop processing the queue
+                    break
+                # Just log everything, filtering will happen on the main process
```

Comment:
This is where we should spend some time, if we know on the parent the desired log level, then all other log messages which don't meet the level requirements can be skipped going through _the wire_.

Reply:
This should already be happening. I'm setting the log level on the ZMQHandler. https://github.com/saltstack/salt/pull/54276/files#diff-feeb9e1b18658ad0ea58ddd2d085bad7R992

Reply:
Awesome.